src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncher.java

```diff
@@ -235,14 +254,18 @@ private void launchPod(String appId, PodSpec podSpec, Map<String, String> labelM
 	}
 
 	private void launchJob(String appId, PodSpec podSpec, Map<String, String> podLabelMap, Map<String, String> idMap,
-						   Map<String, String> annotations) {
+						   Map<String, String> annotations, Map<String, String> jobSpecProperties) {
 		ObjectMeta objectMeta = new ObjectMetaBuilder().withLabels(podLabelMap).addToLabels(idMap).build();
 		PodTemplateSpec podTemplateSpec = new PodTemplateSpec(objectMeta, podSpec);
 
 		JobSpec jobSpec = new JobSpecBuilder()
 				.withTemplate(podTemplateSpec)
 				.build();
 
+		if (jobSpecProperties.containsKey(BACKOFF_LIMIT_KEY)) {
```

Comment:
per above could just get this from `getBackoffLimit(request)`

Reply:
The `request` is not available in this method and also I thought it is not necessary to pass the entire `request` down here.

Reply:
right; would need to be passed in. regardless, i think we should consider consolidating the signatures of `launchJob` and `launchPod` otherwise we will keep tacking on args. a value object could carry say the first 4 args of both methods, then if request is embedded, we wouldn't have to tack on jobAnnotations, this new map, etc

Reply:
Yeah, it makes sense. I have them addressed and updated the PR. Thanks!